binary_and_decimal.rb

```diff
@@ -3,7 +3,21 @@
 #    appears as a binary number. Calculate the decimal value for this binary
 #    number using the algorithm you devised in class. Print out the decimal value.
 def binary_to_decimal
-  puts "Not implemented"
+  binary_number = []
+  8.times do
+    random_no = rand(0..1)
+    binary_number << random_no
+  end
+  decimal_number = []
+  i = binary_number.length
+  binary_number.each do |value|
+    i -= 1
+    number = value * (2 ** i)
+    decimal_number << number
+  end
+  # add all decimal number Array elements together
+  decimal_number = decimal_number.inject(0){|sum,x| sum + x }
+  return decimal_number
```

Comment:
The logic looks good. The problem statement asks you to print both the decimal and binary values in the method. So you're just missing the puts statements.